src/extension.ts

```diff
@@ -420,14 +414,70 @@ export function activate(context: ExtensionContext) {
         xml['xml']['catalogs'].push(catalog);
       }
     })
-    externalXmlSettings.xmlFileAssociations.forEach(element => {
-      if (!xml['xml']['fileAssociations'].some(fileAssociation => fileAssociation.systemId === element.systemId)) {
-        xml['xml']['fileAssociations'].push(element);
-      }
-    });
+    const variableSubstitutedAssociations: XMLFileAssociation[] =
+      xml['xml']['fileAssociations'].map((association: XMLFileAssociation): XMLFileAssociation => {
+
+        const currentFile: TextDocument = window.activeTextEditor.document;
+        const currentFileUri: string = currentFile && currentFile.uri.fsPath;
+        const currentWorkspace: WorkspaceFolder = workspace.getWorkspaceFolder(currentFile && currentFile.uri);
+        const currentWorkspaceUri: string = (currentWorkspace && currentWorkspace.uri.fsPath)
+          || (workspace.workspaceFolders && workspace.workspaceFolders[0].uri.fsPath);
+
+        if (!currentWorkspaceUri
+          && (association.pattern.indexOf('&{workspaceFolder}') >= 0
```

Comment:
I think we should have a global const with variables substitution information (name + referencedCurrentFile) and after that you could use this array to do your logic.

The idea is when we will want to add another variable, we should just update the array.

Reply:
Please let me know if the implementation makes sense.

Reply:
It's better, but for `subVars` it's not generic. The idea is to have just an array that you update and that's it. To fix subVars, I suggest that VariableReference defines a method replace or update which update a given string.  Each VariablerReference defines update method with your current replace. After that `subVars`  will just loop for VariablerReference and call the update/replace method.

Reply:
Thanks so much for the review. The code is much easier to read and add onto now. Please let me know if you think its good when you have time.

Reply:
Glad your code pleases you:) It's perfect.